pkg/client/powervc/powervc.go

```diff
@@ -0,0 +1,483 @@
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/flavors"
+	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/imagedata"
+	"github.com/gophercloud/gophercloud/openstack/imageservice/v2/images"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/networks"
+	"github.com/gophercloud/gophercloud/openstack/networking/v2/subnets"
+	"github.com/gophercloud/utils/openstack/clientconfig"
+
+	"github.com/ppc64le-cloud/hypershift-agent-automation/log"
+	"github.com/ppc64le-cloud/hypershift-agent-automation/util"
+)
+
+type Client struct {
+	computeClient   *gophercloud.ServiceClient
+	imageClient     *gophercloud.ServiceClient
+	networkClient   *gophercloud.ServiceClient
+	volumeClient    *gophercloud.ServiceClient
+	StorageTemplate string
+	NetworkName     string
+	Host            string
+}
+
+type Agent struct {
+	ID            string
+	Name          string
+	PartitionName string
+	IP            string
+	MAC           string
+}
+
+const (
+	DiskSize = 120
+)
+
+var tags = []string{"purpose:hypershift-bm-agent-ci"}
+
+var VolumeNotFound = func(name string) error { return fmt.Errorf("volume not found by the name %s", name) }
+var ImageNotFound = func(name string) error { return fmt.Errorf("image not found by the name %s", name) }
+
+func LoadEnv(c *Client) error {
+	var set bool
+	var errs []error
+
+	c.StorageTemplate, set = os.LookupEnv("POWERVC_STORAGE_TEMPLATE")
+	if !set {
+		errs = append(errs, fmt.Errorf("POWERVC_STORAGE_TEMPLATE env var not set"))
+	}
+	c.NetworkName, set = os.LookupEnv("POWERVC_NETWORK_NAME")
+	if !set {
+		errs = append(errs, fmt.Errorf("POWERVC_NETWORK_NAME env var not set"))
+	}
+	c.Host, set = os.LookupEnv("POWERVC_HOST")
+	if !set {
+		errs = append(errs, fmt.Errorf("POWERVC_HOST env var not set"))
+	}
+
+	return errors.Join(errs...)
+}
+
+func NewClient() (*Client, error) {
+	options := &clientconfig.ClientOpts{}
+	imageClient, err := clientconfig.NewServiceClient("image", options)
+	if err != nil {
+		return nil, err
+	}
+
+	computeClient, err := clientconfig.NewServiceClient("compute", options)
+	if err != nil {
+		return nil, err
+	}
+
+	networkClient, err := clientconfig.NewServiceClient("network", options)
+	if err != nil {
+		return nil, err
+	}
+
+	volumeClient, err := clientconfig.NewServiceClient("volume", options)
+	if err != nil {
+		return nil, err
+	}
+
+	c := &Client{
+		imageClient:   imageClient,
+		computeClient: computeClient,
+		networkClient: networkClient,
+		volumeClient:  volumeClient,
+	}
+
+	if err = LoadEnv(c); err != nil {
+		return nil, fmt.Errorf("error loading env var of powervc: %v", err)
+	}
+
+	return c, nil
+}
+
+func (c Client) GetVolumeIDByName(name string) (string, error) {
+	volumeListPage, err := volumes.List(c.volumeClient, volumes.ListOpts{Name: name}).AllPages()
+	if err != nil {
+		return "", err
+	}
+
+	volumeList := volumeListPage.GetBody().(map[string][]interface{})["volumes"]
```

Comment:
Can't we use ExtractVolume() https://pkg.go.dev/github.com/gophercloud/gophercloud/openstack/blockstorage/v1/volumes#ExtractVolumes, Instead of typecasting